frontend/src/views/CanteensPage/CanteensHome.vue

```diff
@@ -43,54 +157,173 @@
   },
   methods: {
     fetchCurrentPage() {
-      return this.$store.dispatch("fetchPublishedCanteens", { offset: this.offset })
+      let queryParam = `limit=${this.limit}&offset=${this.offset}`
+      if (this.searchTerm) queryParam += `&search=${this.searchTerm}`
+      if (this.appliedFilters.chosenDepartment) queryParam += `&department=${this.appliedFilters.chosenDepartment}`
+      if (this.appliedFilters.minMealCount) queryParam += `&min_daily_meal_count=${this.appliedFilters.minMealCount}`
+      if (this.appliedFilters.maxMealCount) queryParam += `&max_daily_meal_count=${this.appliedFilters.maxMealCount}`
+
+      for (let i = 0; i < this.appliedFilters.chosenSectors.length; i++)
+        queryParam += `&sectors=${this.appliedFilters.chosenSectors[i]}`
+
+      return fetch(`/api/v1/publishedCanteens/?${queryParam}`)
+        .then((response) => {
+          if (response.status < 200 || response.status >= 400) throw new Error(`Error encountered : ${response}`)
+          return response.json()
+        })
+        .then((response) => {
+          this.publishedCanteenCount = response.count
+          this.visibleCanteens = response.results
+        })
+        .catch(() => {
+          this.publishedCanteenCount = 0
+          this.$store.dispatch("notifyServerError")
+        })
+    },
+    clearSearch() {
+      this.searchTerm = ""
+      this.search()
+    },
+    search() {
+      const override = this.searchTerm ? { page: 1, recherche: this.searchTerm } : { page: 1 }
+      const query = Object.assign(this.query, override)
+      this.$router.push({ query }).catch(() => {})
+    },
+    clearFilters() {
+      this.appliedFilters = {
+        chosenDepartment: null,
+        chosenSectors: [],
+        minMealCount: null,
+        maxMealCount: null,
+      }
+    },
+    changePage() {
+      const override = this.page ? { page: this.page } : { page: 1 }
+      const query = Object.assign(this.query, override)
+      this.$router.push({ query }).catch(() => {})
+    },
+    applyFilter() {
+      const changedKeys = Object.keys(getObjectDiff(this.query, this.$route.query))
+      const shouldNavigate = changedKeys.length > 0
+      if (shouldNavigate) this.$router.push({ query: Object.assign(this.query, { page: 1 }) }).catch(() => {})
+      else this.fetchCurrentPage()
+    },
+    populateParameters() {
+      this.page = this.$route.query.page ? parseInt(this.$route.query.page) : 1
+      this.searchTerm = this.$route.query.recherche || null
+      this.appliedFilters = {
+        chosenDepartment: this.$route.query.departement || null,
+        chosenSectors: this.$route.query.secteurs?.split?.("+").map((x) => parseInt(x)) || [],
+        minMealCount: parseInt(this.$route.query.minRepasJour) || null,
+        maxMealCount: parseInt(this.$route.query.maxRepasJour) || null,
+      }
+    },
+    onChangeMealCount(ref) {
+      if (this.$refs[ref].validate()) this.appliedFilters[ref] = parseInt(this.$refs[ref].lazyValue)
```

Comment:
```suggestion
      if (this.$refs[ref].validate()) this.appliedFilters[ref] = parseInt(this.$refs[ref].lazyValue) || null
```
sans ça, si il y a des filtres et on clique sur un min/max pas remplis et tap `enter`, on reçoit `NaN`:

https://user-images.githubusercontent.com/9282816/129598950-cf144d73-1790-4a36-a0d6-eeed0996f195.mov

Reply:
Merci !